libsolidity/analysis/TypeChecker.cpp

```diff
@@ -641,6 +641,14 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 			solAssert(var->type(), "Expected variable type!");
 			if (var->isConstant())
 			{
+				{
+					Identifier const* identifier;
+					VariableDeclaration const* referencedVariable;
+					while ((identifier = dynamic_cast<Identifier const*>(var->value().get())) &&
+						(referencedVariable = dynamic_cast<VariableDeclaration const*>(identifier->annotation().referencedDeclaration)))
+							var = referencedVariable;
+				}
```

Comment:
@chriseth This solves the problem of referencing unsupported constants, but is there a better way to write this block?

Reply:
Looks rather good to me! If you extract that into a function, maybe you can use the same in the ContractCompiler?

Reply:
In which file should I place this function?

Reply:
I think it should be inside the ast folder, maybe a new file `ASTUtils.h/cpp`?

Reply:
Done here https://github.com/ethereum/solidity/pull/7874/commits/b404ae0f7110d92ec0ac852ecf2f9adcb87709f9.